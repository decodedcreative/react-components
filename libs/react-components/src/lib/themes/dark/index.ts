import colors from './colors';
import components from './components';
import general from './general';
import spacing from './spacing';
import typography from './typography';

const darkTheme = {
  name: 'darkTheme',
  colors,
  components,
  general,
  spacing,
  typography,
};

export default darkTheme;
