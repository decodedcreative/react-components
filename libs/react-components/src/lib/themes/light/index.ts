import colors from './colors';
import components from './components';
import general from './general';
import typography from './typography';

const lightTheme = {
  name: 'lightTheme',
  colors,
  components,
  general,
  typography,
};

export default lightTheme;
