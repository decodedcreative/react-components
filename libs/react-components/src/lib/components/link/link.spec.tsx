import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../themes';
import Link, { LinkProps } from './link';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const createComponentJSX = ({
  children = 'Button text',
  decoration = false,
  variant,
}: LinkProps) => (
  <ThemeProvider theme={lightTheme}>
    <Link decoration={decoration} variant={variant}>
      {children}
    </Link>
  </ThemeProvider>
);

const createComponentRender = ({
  children = 'Link text',
  variant,
  decoration,
}: LinkProps) =>
  render(
    <ThemeProvider theme={lightTheme}>
      <Link decoration={decoration} variant={variant}>
        {children}
      </Link>
    </ThemeProvider>
  );

describe('Link', () => {
  it('should render successfully', () => {
    const component = createComponentRender({} as LinkProps);
    expect(component).toBeTruthy();
  });
  describe('Link - Variants', () => {
    it('should match the secondary link snapshot', () => {
      const secondaryLink = createComponentJSX({} as LinkProps);
      const secondaryLinkJSON = renderer.create(secondaryLink).toJSON();
      expect(secondaryLinkJSON).toMatchSnapshot();
    });
    it('should match the inverse link snapshot', () => {
      const inverseLink = createComponentJSX({
        variant: 'inverse',
      } as LinkProps);
      const inverseLinkJSON = renderer.create(inverseLink).toJSON();
      expect(inverseLinkJSON).toMatchSnapshot();
    });
  });
  describe('Link - Options', () => {
    it('should match the link with no text decoration snapshot', () => {
      const noDecorationLink = createComponentJSX({
        decoration: false,
      } as LinkProps);
      const noDecorationLinkJSON = renderer.create(noDecorationLink).toJSON();
      expect(noDecorationLinkJSON).toMatchSnapshot();
    });
  });
});
