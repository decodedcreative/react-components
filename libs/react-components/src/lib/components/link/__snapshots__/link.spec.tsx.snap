// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Link - Options should match the link with no text decoration snapshot 1`] = `
.c0 {
  color: #006de3;
  font-family: 'Source Sans Pro',sans-serif;
  line-height: 1.5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
>
  Button text
</a>
`;

exports[`Link Link - Variants should match the inverse link snapshot 1`] = `
.c0 {
  color: #006de3;
  font-family: 'Source Sans Pro',sans-serif;
  line-height: 1.5;
  cursor: pointer;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
>
  Button text
</a>
`;

exports[`Link Link - Variants should match the secondary link snapshot 1`] = `
.c0 {
  color: #006de3;
  font-family: 'Source Sans Pro',sans-serif;
  line-height: 1.5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
>
  Button text
</a>
`;
